maintenance/provision-lxc-node.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -ex
+source ./maintenance/helpers-lxc-node.sh
+
+container=${NODE_NAME:-"juju-client-box"}
+
+lxc launch ubuntu:16.04 $container
+
+while ! lxc info $container | grep -qE 'eth0:\sinet\s'; do
+    sleep 3
+done
+
+RUN add-apt-repository -y ppa:juju/stable
+RUN apt update -yq
+RUN apt install -y juju unzip python3-pip python-pip charm-tools squashfuse snapd
+RUN pip2 install 'git+https://github.com/juju-solutions/bundletester' \
+                 'git+https://github.com/juju-solutions/cloud-weather-report' \
+                 'git+https://github.com/juju/juju-crashdump'
+RUN pip3 install 'git+https://github.com/juju/python-libjuju' \
+                 'git+https://github.com/juju-solutions/matrix' \
+                 'git+https://github.com/juju/amulet'
+
+
+RUN pip2 install -U charm-tools pyopenssl
+RUN pip3 install -U pytest  pytest-asyncio asyncio_extras juju requests pyyaml kubernetes
+RUN snap install conjure-up --classic
+
+RUN mkdir -p /srv/artifacts
+RUN mkdir -p /root/.local/share/juju
+
+PUSH ~/.local/share/juju/accounts.yaml /root/.local/share/juju/
+PUSH ~/.local/share/juju/models.yaml /root/.local/share/juju/
+PUSH ~/.local/share/juju/controllers.yaml /root/.local/share/juju/
+PUSH ~/.local/share/juju/credentials.yaml /root/.local/share/juju/ 
+RUN mkdir -p /var/lib/jenkins/.local/share/juju
+PUSH ~/.local/share/juju/foo.json /var/lib/jenkins/.local/share/juju/
```

Comment:
Here's foo.json again. Is this a real thing?

Reply:
Lol, yes I saw this today. It is needed by the google provider https://ci.kubernetes.juju.solutions/job/test-cdk/96/console Will digg into this and let you know more soon.

Reply:
That json file includes the gce credentials as described here: https://jujucharms.com/docs/2.1/help-google. The file name is unfortunate.